File: Remove Duplicates from Sorted Array II - Leetcode 80/Remove Duplicates from Sorted Array II - Leetcode 80.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int j = 1;
        int count = 1;
        int n = nums.length;

        for (int i = 1; i < n; i++) {
            if (nums[i] == nums[i - 1]) {
                count++;
            } else {
                count = 1;
            }

            if (count <= 2) {
                nums[j] = nums[i];
                j++;
            }
        }

        return j;
    }
}

File: Remove Duplicates from Sorted Array II - Leetcode 80/Remove Duplicates from Sorted Array II - Leetcode 80.js
var removeDuplicates = function(nums) {
    var j = 1;
    var count = 1;
    var n = nums.length;

    for (var i = 1; i < n; i++) {
        if (nums[i] === nums[i - 1]) {
            count++;
        } else {
            count = 1;
        }

        if (count <= 2) {
            nums[j] = nums[i];
            j++;
        }
    }

    return j;
};
